build_tools/update_iree_version_pins.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+# Copyright 2025 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+"""Updates the pinned IREE package versions in this repository.
+
+Usage:
+  update_iree_version_pins.py
+"""
+
+from pathlib import Path
+import re
+import subprocess
+import sys
+import textwrap
+
+REPO_ROOT = Path(__file__).parent.parent
+REQUIREMENTS_IREE_PINNED_PATH = REPO_ROOT / "requirements-iree-pinned.txt"
+
+
+def get_latest_package_version(package_name, extra_pip_args=[]):
+    print("\n-------------------------------------------------------------------------")
+    print(f"Finding latest available package version for package '{package_name}'\n")
+
+    # This queries the pip index to get the latest version.
+    #
+    # Note: the `index` subcommand is experimental. Other possible approaches:
+    #   * Install (into a venv) then check what was installed with --report or 'pip freeze'
+    #   * Download then check what was downloaded
+    #   * Scrape the package index and/or release page (https://iree.dev/pip-release-links.html)
```

Comment:
* https://stackoverflow.com/questions/71625061/getting-pip-dependency-versions-without-installing
* https://stackoverflow.com/questions/75667225/how-to-get-latest-version-of-not-installed-package-using-pip